loader.js

```diff
@@ -1,16 +1,20 @@
 const requireFromString = require('require-from-string');
 const renderer = require('react-dom/server');
-const NODE_ENV = require('process').env.NODE_ENV;
+const { getOptions } = require('loader-utils');
 
 module.exports = function loader(source) {
   let code = source;
-  if (NODE_ENV !== 'development') {
-      code = code.replace(/condition: (.*)?([\s\,])/, 'condition: "$1"$2');
+  const { env } = getOptions(this);
+  if (env !== 'development') {
+    code = code.replace(/condition: (.*)?([\s\,])/, 'condition: "$1"$2');
   }
   console.log('---- render component ----')
   console.log('code\n', code);
   console.log('resourcePath', this.resourcePath);
-  const component = requireFromString(code, this.resourcePath);
+  const module = requireFromString(code, this.resourcePath); // transforms the received string into actual code and returns its output
+  console.log('module', module);
+  const component = module(env);
+  console.log('component', component);
```

Comment:
Can we use `logger` for all those messages ?

Reply:
Yes, sure!